### DISCUSSION TITLE
How do you add a chain to the WAGMI config at runtime?

### DISCUSSION THREAD

**Discussion by nnn-gif** [wevm/wagmi#4358] [General]
Currently, the WAGMI config requires the chain to be passed during initialization. Is there a way to add more chains on the fly?

**Comment by tmm** [comment]
You can use `config._internal.chains.setState`. To see how it's used, you can [search the repo for examples.](https://github.com/search?q=repo%3Awevm%2Fwagmi+_internal.chains.setState&type=code). Keep in mind that `config._internal` is not covered by semver and may have breaking changes at any time without notice.
